tests/fracs/test_fracture_network_2d.py

```diff
@@ -0,0 +1,337 @@
+        assert compare_arrays(p_known, together._pts)
+        # The known edges has 2 rows, thus by testing for equality, we implicitly
+        # verify there are no tags in the joint network
+        e_known = np.array([[0, 2, 4, 6], [1, 3, 5, 7]])
+        assert compare_arrays(together._edges, e_known)
+
+    def test_add_networks_domains(self):
+        network_1 = pp.create_fracture_network(self.fracs, self.domain)
+        p2 = np.array([[0, 2, 1, 1], [0, 0, 0, 1]]) + 2
+        e2 = np.array([[0, 2], [1, 3]])  #
+        fracs2 = pts_edges_to_linefractures(p2, e2)
+        # A network with no domain
+        network_2 = pp.create_fracture_network(fracs2)
+
+        # Add first to second, check domain
+        together = network_1.add_network(network_2)
+        assert self.compare_dictionaries(
+            self.domain.bounding_box, together.domain.bounding_box
+        )
+
+        # Add second to first, check domain
+        together = network_2.add_network(network_1)
+        assert self.compare_dictionaries(
+            self.domain.bounding_box, together.domain.bounding_box
+        )
+
+        # Assign domain, then add
+        network_2.domain = self.domain
+        together = network_1.add_network(network_2)
+        assert self.compare_dictionaries(
+            self.domain.bounding_box, together.domain.bounding_box
+        )
+
+        # Assign different domain, check that the sum has a combination of the
+        # domain dicts
+        network_2.domain = self.small_domain
+        together = network_1.add_network(network_2)
+        combined_domain = pp.Domain({"xmin": -1, "xmax": 5.0, "ymin": -1, "ymax": 5})
+        assert self.compare_dictionaries(
+            combined_domain.bounding_box, together.domain.bounding_box
+        )
+
+    def test_add_networks_preserve_tags(self):
+        network_1 = pp.create_fracture_network(self.fracs)
+        p2 = np.array([[0, 2, 1, 1], [0, 0, 0, 1]]) + 2
+        # Network 2 has tags
+        tag2 = 1
+        e2 = np.array([[0, 2], [1, 3], [1, 1]])
+        fracs2 = pts_edges_to_linefractures(p2, e2)
+        network_2 = pp.create_fracture_network(fracs2)
+
+        # Add networks, check tags
+        together = network_1.add_network(network_2)
+        assert np.all(together._edges[2, 2:4] == tag2)
+
+        # Add networks reverse order
+        together = network_2.add_network(network_1)
+        assert np.all(together._edges[2, :2] == tag2)
+
+        # Assign tags to network1
+        tag1 = 2
+        # Had to change this to ensure that edges still has ``dtype np.int8``
+        network_1._edges = np.vstack(
+            (network_1._edges, tag1 * np.ones(2, dtype=np.int8))
+        )
+        together = network_1.add_network(network_2)
+        known_tags = np.array([tag1, tag1, tag2, tag2])
+        assert np.all(together._edges[2] == known_tags)
+
+    # Below are tests with polytopal domains
+
+    def test_triangle_domain_no_fracs(self):
```

Comment:
The following three tests could be parametrized - correct?

Reply:
Changing it